core/Filename_.mli

```diff
@@ -0,0 +1,35 @@
+(*
+   A trivial wrapper around "string" to make signatures clearer.
+   For internal use only.
+
+   It would be nicer to use 'Fpath' but we're trying to minimize
```

Comment:
meh, I think it's fine to depend on fpath.

Reply:
For now, it makes fewer things to worry about. If we need a complex function such as `Fpath.normalize`, it would probably be a good move to use `Fpath` rather than reimplement our own.